eng/versioning/verify_release_set.ps1

```diff
@@ -81,11 +84,25 @@ if ($ArtifactsList -and $ArtifactsList.Count -gt 0) {
                         {
                             $libraryToVerify = "$($groupId):$($artifactId)"
                             if (!$librariesToRelease.ContainsKey($libraryToVerify)) {
-                                if (!$missingLibraries.ContainsKey($libraryToVerify)) {
-                                    $missingLibraries[$libraryToVerify] = $true
+                                # Only check whether or not the library's version has been published if
+                                # it hasn't already been checked
+                                $releasedLibToVerify = "$($libraryToVerify):$($version)"
+                                Write-Host "$releasedLibToVerify is a current dependency of $($inputGroupId):$($inputArtifactId) and is not on the release list, checking if $version has been published."
+                                if (!$mavenPublished.ContainsKey($releasedLibToVerify)) {
+                                    $released = IsMavenPackageVersionPublished -pkgId $artifactId -pkgVersion $version -groupId $groupId
+                                    $mavenPublished.Add($releasedLibToVerify, $released)
+                                }
+                                # If the current dependency isn't on the command line to be released and the version
+                                # isn't published on Maven, then report an error
+                                if (!$mavenPublished[$releasedLibToVerify]) {
+                                    if (!$missingLibraries.ContainsKey($libraryToVerify) -and !$mavenPublished) {
```

Comment:
```suggestion
                                    if (!$missingLibraries.ContainsKey($libraryToVerify)) {
```

Not sure why we needed that mavenPublished check as it always is non-null.

Reply:
Also gone. That was going to be something different before I'd simplified it.